File: E1/practice1/garden/Garden.java
import java.util.ArrayList;

public class Garden {
    public Garden(Light light) {
        this.light = light;
        this.plants = new ArrayList<>();
    }
    public void addPlant(Plant plant) {
        plants.add(plant);
    }
    @Override
    public String toString() {
        String result = "";
        String separator = "";
        for(Plant p : plants) {
            result += separator + p;
            separator = ", ";
        }
        return result;
    }
    private ArrayList<Plant> plants;
    private Light light;
}

File: E3/practice1/cpp/garden.cpp
#include "garden.h"

Garden::Garden(Light light) : _light{light} { }
void Garden::add_plant(Plant plant) {_plants.push_back(plant);}
std::string Garden::to_string() {
    std::string s;
    for(Plant p : _plants) {
        s += p.name() + " (" + ::to_string(p.light()) + "), ";
    }
    return s;
}

